app/create.py

```diff
@@ -287,6 +274,112 @@ def create_person(
+    if status == 'is_note_author' and not author_made_contact:
+        raise CreationError(_(
+            'Please check that you have been in contact with the person after '
+            'the disaster, or change the "Status of this person" field.'))
+    if status == 'believed_dead' and not config.allow_believed_dead_via_ui:
+        raise CreationError(_(
+            'Not authorized to post notes with the status "believed_dead".'))
+    if author_email and not validate_email(author_email):
+        raise CreationError(_(
+            'The email address you entered appears to be invalid.'))
+
+
+def create_note(
+        repo,
+        person,
+        config,
+        user_ip_address,
+        status,
+        source_date,
+        author_name,
+        author_email,
+        author_phone,
+        author_made_contact,
+        photo,
+        photo_url,
+        text,
+        email_of_found_person,
+        phone_of_found_person,
+        last_known_location,
+        validate_data=True):
+    if validate_data:
+        validate_note_data(
+            config=config,
+            status=status,
+            author_name=author_name,
+            author_email=author_email,
+            author_made_contact=author_made_contact,
+            text=text)
+    spam_detector = SpamDetector(config.bad_words)
+    spam_score = spam_detector.estimate_spam_score(text)
+    if spam_score > 0:
+        note = NoteWithBadWords.create_original(
+            repo,
+            entry_date=get_utcnow(),
+            person_record_id=person.record_id,
+            author_name=author_name,
+            author_email=author_email,
+            author_phone=author_phone,
+            source_date=source_date,
+            author_made_contact=author_made_contact,
+            status=status,
+            email_of_found_person=email_of_found_person,
+            phone_of_found_person=phone_of_found_person,
+            last_known_location=last_known_location,
+            text=text,
+            photo=photo,
+            photo_url=photo_url,
+            spam_score=spam_score,
+            confirmed=False)
+        note.put_new()
+        raise FlaggedNoteException(note)
+    else:
+        note = Note.create_original(
+            repo,
+            entry_date=get_utcnow(),
+            person_record_id=person.record_id,
+            author_name=author_name,
+            author_email=author_email,
+            author_phone=author_phone,
+            source_date=source_date,
+            author_made_contact=author_made_contact,
+            status=status,
+            email_of_found_person=email_of_found_person,
+            phone_of_found_person=phone_of_found_person,
+            last_known_location=last_known_location,
+            text=text,
+            photo=photo,
+            photo_url=photo_url)
+        note.put_new()
+        # Specially log notes that make a person dead or switch to an alive
+        # status.
+        if status == 'believed_dead':
```

Comment:
just a generic comment on hard-coded string statuses vs constants if that's relevant